framework/db/ActiveRecord.php

```diff
@@ -455,18 +455,28 @@ protected function insertInternal($attributes = null)
             }
         }
         $db = static::getDb();
-        $command = $db->createCommand()->insert($this->tableName(), $values);
-        if (!$command->execute()) {
+        // depending on the underlying DBMS, the command may return results
+        // or lastInsertId has to be used to fetch primary key values
+        $command = $db->createCommand()->insertReturning($this->tableName(), $values, $this->primaryKey());
+        // force preparing as a command for writing, not reading
+        $command->prepare(false);
+        $primaryKeys = $command->queryOne();
```

Comment:
That's why the base `QueryBuilder` returns a normal `INSERT` ignoring `returnColumns`.

If you execute an `INSERT` statement in MySQL and then try to fetch results from the PDO statement it throws a general error. There's a check in `Command::queryInternal` so it doesn't do that, making it behave like `execute`.